avaframe/out1Peak/outPlotAllPeak.py

```diff
@@ -68,11 +74,22 @@ def plotAllPeakFields(avaDir, cfg, cfgFLAGS, modName='com1DFA'):
         # Load data
         raster = IOf.readRaster(fileName)
         data = raster['rasterData']
-        data = np.ma.masked_where(data == 0.0, data)
+
+        # constrain data to where there is data
+        cellSize = peakFiles['cellSize'][m]
+        ind = np.where(data>0)
+        plotBuffer = int(pU.cfg.getfloat('plotBuffer') / cellSize)
+        rowsMin = max(np.amin(ind[0])-plotBuffer, 0)
+        rowsMax = min(np.amax(ind[0])+plotBuffer, data.shape[0])
+        colsMin = max(np.amin(ind[1])-plotBuffer, 0)
+        colsMax = min(np.amax(ind[1])+plotBuffer, data.shape[1])
+        dataConstrained = data[rowsMin:rowsMax+1, colsMin:colsMax+1]
+        demConstrained = demField[rowsMin:rowsMax+1, colsMin:colsMax+1]
+
+        data = np.ma.masked_where(dataConstrained == 0.0, dataConstrained)
```

Comment:
it would be interesting to have this in a separate function that we can use in other plots!!